include/lin/generators/randoms.hpp

```diff
@@ -51,10 +52,21 @@ class RandomsGenerator {
 
     return 5.42101086242752217E-20 * (seed * 2685821657736338717ULL);
   }
+
+  /**
+   * @brief Return a gaussian random number with mean 0 and std 1. Uses Box–Muller transform.
+   * 
+   * @return constexpr double 
+   */
+  constexpr double gaussian() {
+    double R = sqrt(-2.0L*log(next()));
+    double T = 2.0L*M_PI*next();
+    return R*cos(T);
+  }
```

Comment:
Ah so given the code we have for `gaussian_rands` could we use `R * sin(T)` as another random sample as well? If so, I think we should add a field to the `RandomsGenerator` that will cache this to return on a subsequent call to `gaussian` in order reduce overhead from these calls to math functions. This would also make the implementation of `gaussian_rands` much simpler and centralize the Box-Muller transform.

Secondly, we should prefix the math calls with `std::`. We've had issues with using math function from `math.h` in the past (don't really understand why but weird stuff was happening on Teensy).

Reply:
Added caching and std::